application/modules/book_receive/controllers/Book_receive.php

```diff
@@ -465,9 +465,22 @@ public function final($book_receive_id = null, $action = null)
             'finish_date' => $action == 'finish' ? now() : null
         ]);
 
-        // update book stock
         $book_stock = $this->book_stock->where('book_id', $book_receive->book_id)->get();
         $book_stock_print = $this->book_receive->get_print_order($book_receive->print_order_id);
+
+        //insert to book transaction
+        $book_stock = $this->book_stock->where('book_id', $book_receive->book_id)->get();
```

Comment:
double `$book_stock`

ketika klik finalisasi penerimaan buku, maka muncul error seperti gambar di bawah.
oiya kenapa perlu `/finish` di belakang? bukannya kalo finalisasi book receive udah pasti statusnya di set ke `finish`? atau ada kondisi lain?

![image](https://user-images.githubusercontent.com/14112952/117935470-ef9ee400-b32d-11eb-8da6-c6285d961937.png)

Reply:
oh iya mas nanti dihapus `$book_stock` yang double.
harusnya tidak perlu /finish sih mas karena emang status akhirnya selalu finish, aku ganti ya mas biar langsung /final/id aja.
di tempatku bisa mas, mungkin karena di database mas bagas ada beberapa baris yang book_id nya dobel/sama? karena harusnya book sama book_stock relasinya one to one. mungkin bisa dihapus dulu mas data yang book_id nya ke-double

Reply:
@esterinac nah data book_stock itu ditambahkan ketika apa?

Reply:
saat final mas

Reply:
coba cek method `final` di controller `book_request`, kayaknya urutannya kebalik. insert `book_transaction` harusnya setelah pengecekan `book_stock` kan?

cara reproduce errornya gini, coba order cetak buku yang belum ada di book stock.

Reply:
harusnya yg bener kayak gini bukan?

https://github.com/bagaskarala/sigap/blob/develop/application/modules/book_receive/controllers/Book_receive.php#L469-L488

Reply:
oh iya mas kalau cetak buku yang belum ada di book stock jadinya error, karena belum ada `book_stock_id` nya ya. iya mas kebalik ternyata bener yang sebelumnya, makasih mas. sudah kubenerin lagi